### ISSUE TITLE
3 read error detection jobs failed

### ISSUE THREAD

**Issue by frsepulveda** [marbl/canu#2213]
Hi,

I was running canu v2.2 and a found a problem: 
```
In gatekeeper store '/home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/mazzalam.gkpStore':
--   Found 7431380 reads.
--   Found 93509938955 bases (865.83 times coverage).
--
--   Read length histogram (one '*' equals 24507.55 reads):
--        0    999      0
--     1000   1999 113943 ****
--     2000   2999 112703 ****
--     3000   3999 113816 ****
--     4000   4999 112708 ****
--     5000   5999 126439 *****
--     6000   6999 136468 *****
--     7000   7999 125160 *****
--     8000   8999 117085 ****
--     9000   9999 114007 ****
--    10000  10999 177804 *******
--    11000  11999 1493736 ************************************************************
--    12000  12999 1715529 **********************************************************************
--    13000  13999 946414 **************************************
--    14000  14999 849648 **********************************
--    15000  15999 600030 ************************
--    16000  16999 308962 ************
--    17000  17999  80888 ***
--    18000  18999  28550 *
--    19000  19999  17679
--    20000  20999  12504
--    21000  21999   9678
```
................................. Next .....................................

I've been struggling with the the assembly portion of a canu run that is giving me the following error:
```
  /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 33638 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.033638.out 2>&1
    /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 33639 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.033639.out 2>&1
    /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 33640 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.033640.out 2>&1

-- Finished on Wed Mar 15 01:31:13 2023 (3533 seconds) with 19288.374 GB free disk space
----------------------------------------
--
-- 3 read error detection jobs failed:
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--
-- read error detection attempt 2 begins with 33637 finished, and 3 to compute.
----------------------------------------
-- Starting concurrent execution on Wed Mar 15 01:31:15 2023 with 19288.374 GB free disk space (3 processes; 8 concurrently)

    /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 10000 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.010000.out 2>&1
    /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 20000 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.020000.out 2>&1
    /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.sh 30000 > /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/red.030000.out 2>&1

-- Finished on Wed Mar 15 01:31:15 2023 (lickety-split) with 19288.374 GB free disk space
----------------------------------------
--
-- 3 read error detection jobs failed:
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--   job /home/rhasbun/Pancho/DNA/PacBio/1A1/mazzalam-pacbio/unitigging/3-overlapErrorAdjustment/0000.red FAILED.
--
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.

canu failed with 'failed to detect errors in reads.  Made 2 attempts, jobs still failed'.
```
I already try the solution given in "read error detection jobs failed #269" with 
redMemory=8  (then 16 and even 32).
oeaMemory=4 (8 and 16).

but nothing work :( 

here is the command i'm issuing:
canu -assemble -p mazzalam -d mazzalam-pacbio genomeSize=108m -pacbio-raw Species.1A1.subreads.fastq.gz useGrid=false

**Comment by skoren** [comment]
Your command is incorrect. You have raw pacbio reads but are asking canu to assemble these directly which isn't going to work. What are these actual reads? Their lengths seem consistent with HiFi but you're specifying them as raw and they are named subreads which are normally not HiFi. If these are raw reads, don't specify `-assemble` and if they're HiFi provide them as `-pacbio-hifi`. 

It also seems suspicious that you have so much coverage, Canu 2.2 should subset that down to a lower unless you request it not to.

**Comment by frsepulveda** [comment]
> 

Hi skoren! thanks for the advice. They are not pacBio hifi, we just have a lot of coverage. I'll try with just using "canu" and not "canu-assemble".

**Comment by skoren** [comment]
What version does you canu install report, you still don't want to assemble 800x coverage and likely want to subset to 200x or so.

**Comment by frsepulveda** [comment]
I dont know why my version is this: 

Canu v0.0 (+0 commits) r0 unknown-hash-tag-no-repository-available.

My canu was installed via conda using: 

conda install -c conda-forge -c bioconda -c defaults canu

how can i subset my samples to a 200x coverage. I tried with "maxInputCoverage"  and "readSamplingCoverage" but dont work.

**Comment by skoren** [comment]
I'm pretty sure you do not have v2.2, we haven't had a gkpStore for years which your log shows. I confirmed on both linux/OSX when I installed canu using conda with your command it correctly reports v2.2. On OSX I had to force it to use 2.2, it tried to install 1.4 for some reason otherwise which did report the version incorrectly so I'd guess that is the version you have installed. 

I'd suggest downloading the 2.2 pre-compiled binary from GitHub release page and then run with that. It should include maxInputCoverage but it should not even be needed as the default max coverage is 200x.
